src/librbd/ExclusiveLock.cc

```diff
@@ -714,8 +714,11 @@ void ExclusiveLock<I>::handle_shutdown(int r) {
   }
 
   m_image_ctx.aio_work_queue->unblock_writes();
-  m_image_ctx.image_watcher->flush(util::create_context_callback<
-    ExclusiveLock<I>, &ExclusiveLock<I>::complete_shutdown>(this));
+  if (m_image_ctx.image_watcher == nullptr)
```

Comment:
What is this change? If you don't have an image watcher, you shouldn't have an ExclusiveLock either.